rust/src/util.rs

```diff
@@ -84,13 +84,67 @@ impl Display for Sizing {
 }
 
 #[derive(Clone, Eq, Debug)]
-#[cfg_attr(feature = "serde", derive(Serialize, Deserialize), serde(crate = "serde_crate"))]
 pub struct Variant {
     pub name: VariantName,
     pub tag: u8,
 }
 impl_strict_struct!(Variant, STRICT_TYPES_LIB; name, tag);
 
+#[cfg(feature = "serde")]
+mod _serde {
+    use std::str::FromStr;
+
+    use serde_crate::ser::SerializeStruct;
+    use serde_crate::{Deserialize, Deserializer, Serialize, Serializer};
+
+    use super::*;
+
+    impl Serialize for Variant {
+        fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+        where S: Serializer {
+            if serializer.is_human_readable() {
+                serializer.serialize_str(&format!("{}:{}", self.name, self.tag))
+            } else {
+                let mut s = serializer.serialize_struct("Variant", 2)?;
+                s.serialize_field("name", &self.name)?;
+                s.serialize_field("tag", &self.tag)?;
+                s.end()
+            }
+        }
+    }
+
+    impl<'de> Deserialize<'de> for Variant {
+        fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+        where D: Deserializer<'de> {
+            if deserializer.is_human_readable() {
+                let s = String::deserialize(deserializer)?;
+                let parts: Vec<&str> = s.split(':').collect();
```

Comment:
This will really slow down parsing, since TypeSystem may have thousands and thousands of these. No need for allocation, just do 

```rust
let split = s.split(':');
let (name, tag, none) = (split.next(), split.next(), split.next()); 
if none.is_some() { return Err(...) }
```

Reply:
Ok, done